lib/streamlit/runtime/state/widgets.py

```diff
@@ -214,17 +240,7 @@ def register_widget(
                 user_key,
             )
         )
-
-    # Create the widget's updated metadata, and register it with session_state.
-    metadata = WidgetMetadata(
-        widget_id,
-        deserializer,
-        serializer,
-        value_type=ELEMENT_TYPE_TO_VALUE_TYPE[element_type],
-        callback=on_change_handler,
-        callback_args=args,
-        callback_kwargs=kwargs,
-    )
+    caching.save_widget_metadata(metadata)
```

Comment:
Also, I might have just overlooked it, but is this behavior tested anywhere?

Reply:
Not by unit tests, but its role in making replay work is tested by some cypress tests.